operatorcourier/verify.py

```diff
@@ -0,0 +1,143 @@
+import os
+import logging
+from typing import List, Dict
+import semver
+from operatorcourier.build import BuildCmd
+from operatorcourier.validate import ValidateCmd
+from operatorcourier.errors import OpCourierBadBundle
+from operatorcourier.format import format_bundle
+from operatorcourier import identify
+
+logger = logging.getLogger(__name__)
+FLAT_KEY = '__flat__'
+
+
+class Verification:
+    def __init__(self, source_dir, yamls, ui_validate_io, repository):
+        manifests = self.get_manifests_info(source_dir, yamls)
+        self.nested = len(manifests) > 1
+        self.bundle_dict = None
+        self.validation_dict = \
```

Comment:
Why do we need to pass the validation_dict around at all?

Reply:
@kevinrizza Could you please elaborate on your concern?

Reply:
We just use that function to write to a file. Why do we need to return it? Can't we just write to the file in the `build_and_verify()` method? What does the `push()` method need it for at all?

Reply:
@kevinrizza  I'm not sure if I follow your point. Do you want me to move all the code in `get_validation_dict_from_manifests()` method to `build_and_verify()`?

Reply:
What? No. I'm saying that the purpose of the `validation_dict` was explicitly just to optionally print those logs to a file, right? So if we are mutating that object to do other things, that seems wrong. And if we aren't, we don't need to return it we can just do the file write in `build_and_verify()`

Reply:
@kevinrizza But we do need to "mutate" the `validation_dict` which is initialized as an empty dict object with `warnings` and `errors` key. Previously, there is only 1 `validation_dict` because there's only 1 bundle. Now we have multiple bundles and therefore we have multiple those `validation_dict`s. The purpose of `get_validation_dict_from_manifests` method is to combine all those dicts into one single `validation_dict`, so that `build_and_verify` can write the whole thing into one file.

Reply:
We can still do this without explicitly returning it in the object. Just make the field effectively "private" and then create a method on the class that takes a path and writes the validation to a file. Then call that method inside `build_and_verify()`

Reply:
Thanks @kevinrizza . Updated.